src/lib/models/ConfigModel.ts

```diff
@@ -1,172 +1,198 @@
+	result += `${toYamlComment(topLevelDescription.fieldDescriptions!.rules, styled)}\n`;
+	result += `${toYamlRules(relevantEngines, userContext, defaultContext, styled)}\n`;
+	result += '\n';
 
+	// Then the `engines` section
+	result += `${toYamlComment(topLevelDescription.fieldDescriptions!.engines, styled)}\n`;
+	result += `${toYamlEngines(relevantEngines, userContext, styled)}`;
 
-	const value: string = actualValue === defaultValue
-		? `null # ${getMessage(BundleName.ConfigModel, 'template.last-calculated-as', [actualValue])}`
-		: actualValue;
+	return result;
+}
 
-	return `${annotation}\n`
-		+ `${propertyName}: ${value}\n`;
+function getYamlSectionWrapper(): string {
+	return '='.repeat(70);
 }
 
-function yamlFormatRuleSelection(ruleSelection: RuleSelection): string {
-	// Get the annotation and prepend each line with '# ' to make it a YAML comment.
-	const annotation: string = getMessage(BundleName.ConfigModel, `annotation.rules`)
-		.replace(/^./gm, s => `# ${s}`);
+function toYamlComment(comment: string, styled: boolean, indentLength: number = 0): string {
+	// At the start of the string, and at the start of every line...
+	return comment.replace(/^.*/gm, s => {
+		// ...Inject a `# ` at the start to make it a comment...
+		let commented = `# ${s}`;
+		// ...Apply styling if requested...
+		commented = styled ? makeGrey(commented) : commented;
+		// ...And indent to the requested amount
+		return indent(commented, indentLength);
+	});
+}
 
-	if (ruleSelection.getCount() === 0) {
-		return `${annotation}\n`
-			+ `rules: {} # Remove this empty object {} when you are ready to specify your first rule override`;
+function toYamlWithDerivedValueComment(userValue: string, defaultValue: string, styled: boolean): string {
+	if (userValue == null || userValue === defaultValue) {
+		const comment = getMessage(BundleName.ConfigModel, 'template.last-calculated-as', [defaultValue]);
+		return `null ${toYamlComment(comment, styled)}`;
+	} else {
+		return `${userValue}`;
 	}
+}
 
-	let results = annotation + '\n'
-		+ 'rules:\n';
-
-	for (const engine of ruleSelection.getEngineNames()) {
-		results += `  ${engine}:\n`;
-		for (const rule of ruleSelection.getRulesFor(engine)) {
-			const ruleName: string = rule.getName();
-			const severity = `${rule.getSeverityLevel()}`;
-			const tags = rule.getTags();
-			const tagString = tags.length > 0 ? `["${tags.join('", "')}"]` : `[]`;
-			// This section is built by hand instead of using a message, because the message catalog dislikes indentation.
-			results += `    "${ruleName}":\n`
-				+ `      severity: ${severity}\n`
-				+ `      tags: ${tagString}\n`;
+function toYamlRules(relevantEngines: Set<string>, userContext: ConfigContext, defaultContext: ConfigContext, styled: boolean): string {
+	if (userContext.rules.getCount() === 0) {
+		const comment = getMessage(BundleName.ConfigModel, 'template.yaml.no-rules-selected');
+		return `rules: {} ${toYamlComment(comment, styled)}`;
+	}
```

Comment:
This will prevent the config from being shown when disable_engine: true... so I think you need to remove this condition.